aiohttp_wsgi/wsgi.py

```diff
@@ -96,36 +97,41 @@ def __init__(self, inbuf_overflow, max_request_body_size, loop, executor):
         self._content_length = 0
         self._overflow = False
 
-    async def _run(self, fn, *args):
+    @asyncio.coroutine
+    def _run(self, fn, *args):
         if self._overflow:
-            return await self._loop.run_in_executor(self._executor, fn, *args)
+            return (yield from self._loop.run_in_executor(self._executor, fn, *args))
         else:
             return fn(*args)
 
-    async def write(self, data):
+    @asyncio.coroutine
+    def write(self, data):
         self._content_length += len(data)
         # Check for body size overflow. The request might be streaming, so we check with every chunk.
         if self._content_length > self._max_request_body_size:
             raise HTTPRequestEntityTooLarge()
         # Overflow onto disk, if required.
         if not self._overflow and self._content_length > self._inbuf_overflow:
             self._overflow = True
-            overflow_body = await self._run(TemporaryFile)
-            await self._run(overflow_body.write, self._body.getbuffer())
+            overflow_body = yield from self._run(TemporaryFile)
+            yield from self._run(overflow_body.write, self._body.getbuffer())
             self._body.close()
             self._body = overflow_body
         # Write the block.
-        await self._run(self._body.write, data)
+        yield from self._run(self._body.write, data)
 
-    async def get_body(self):
-        await self._run(self._body.seek, 0)
+    @asyncio.coroutine
+    def get_body(self):
+        yield from self._run(self._body.seek, 0)
         return self._body, self._content_length
 
-    async def __aenter__(self):
+    @asyncio.coroutine
+    def __aenter__(self):
         return self
 
-    async def __aexit__(self, *exc_info):
-        await self._run(self._body.close)
+    @asyncio.coroutine
+    def __aexit__(self, *exc_info):
+        yield from self._run(self._body.close)
```

Comment:
Lose __aenter__ and __aexit__, since Python 3.4 doesn't support async with.

Replace them with a close() coroutine, and call that in a try/finally instead of the async with, below.